IQSS/dataverse-frontend
# Feat/561 add collection sort

## What this PR does / why we need it:
Adds the Sort button to the Collection page, for sorting collection items.

## Which issue(s) this PR closes:

- Closes #561 

## Special notes for your reviewer:

In the Story and the ItemsSortBy component test, the style is  not being applied to the selected option, even though it works in the app and in the CollectionItemsPanel Story.

## Suggestions on how to test this:
**Step 1: Run the Development Environment**
1. Execute `npm i`.
2. Navigate with `cd packages/design-system && npm i && npm run build`.
3. Return with `cd ../../`.
4. Ensure you have a `.env` file similar to `.env.example`, with the variable `VITE_DATAVERSE_BACKEND_URL=http://localhost:8000`.
5. Navigate with `cd dev-env`.
6. Start the environment using `./run-env.sh unstable`.
7. To verify the environment, visit <http://localhost:8000> and check your local Dataverse installation.

**Step 2: Test the feature**
1. Create some collections and datasets in the root collection. Add one or multiple files in one of those created datasets.
2. When first navigating to `http://localhost:8000/spa/collections`, check that the Sort button has ' Newest' highlighted, and the 'Relevance' sort option doesn't exist.
3. Choose another sorting option.  Check that the url is updated with the correct sort parameters.
4. Submit a search.  Check that the sort button now has 'Relevance' highlighted, and the url only has the 'q=searchtext' param.
5. Choose another sorting option.  Check that 'Relevance' is still visible, but not highlighted. Check that the items are sorted in the correct order.



## Does this PR introduce a user interface change? If mockups are available, please link/include them here:

## Is there a release notes update needed for this change?:

## Additional documentation:
